aws-xray-propagator/src/main/java/io/opentelemetry/contrib/awsxray/propagator/AwsXrayPropagator.java

```diff
@@ -112,49 +108,20 @@ public <C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> se
     char samplingFlag = spanContext.isSampled() ? IS_SAMPLED : NOT_SAMPLED;
     // TODO: Add OT trace state to the X-Ray trace header
 
-    StringBuilder traceHeader = new StringBuilder();
-    traceHeader
-        .append(TRACE_ID_KEY)
-        .append(KV_DELIMITER)
-        .append(xrayTraceId)
-        .append(TRACE_HEADER_DELIMITER)
-        .append(PARENT_ID_KEY)
-        .append(KV_DELIMITER)
-        .append(parentId)
-        .append(TRACE_HEADER_DELIMITER)
-        .append(SAMPLED_FLAG_KEY)
-        .append(KV_DELIMITER)
-        .append(samplingFlag);
-
-    Baggage baggage = Baggage.fromContext(context);
-    // Truncate baggage to 256 chars per X-Ray spec.
```

Comment:
What do you think about this [comment](https://github.com/aws-observability/aws-otel-java-instrumentation/issues/563#issuecomment-1924450848):

> Regarding the length of the X-Ray header, I did not find a definition in X-Ray spec. However, based on some clues, it can be inferred that StepFunction likely has a contract with X-Ray, hence there is length checking in the StepFunction client API. So it is reasonable to truncate the baggage length to (256 - the length of the existing trace context).

My read of this is that including up to 256 baggage bytes is a bug that should be fixed, instead of removing baggage encoding altogether.

Reply:
Right. [#563](https://github.com/aws-observability/aws-otel-java-instrumentation/issues/563) is StepFunction receives a xray header over 256. AWS Lambda is working on a new encoding algorithm to reducing the xray header to be less than 256, then we can fix the bug by truncating xray header to be 256, not truncating baggage to 256.